### PR TITLE
Add the firework for calculating the electronic energy of a proton in different solvent environments[WIP]

### PR DESCRIPTION
**Pull request by rdguha1995**
## Summary

For this custom Firework the electronic energy of a proton in a specific solvent environment is approximated. Since a proton has 0 electrons, running a QChem job would yield an error. The energy can be approximated by:

* Calculating the energy of a hydrogen atom with 0 charge
* Calculating the energy of a system with a hydrogen atom (0 charge) and a proton separated at infinite distance 
* Subtracting the first energy from the second

This Firework combines these two calculations and adds a task doc to the DB with the separate calculation details and the effective energy after subtraction.

## TODO (if any)

* Should be reviewed by @samblau before merge 
* Needs a CI test run.

### CODE REVIEW

**Comment by samblau** [comment]
@Zhuoying, please trigger the tests here

**Comment by Zhuoying** [comment]
Thanks @samblau for the reminder. -Just trigger tests and it looks like passing the tests. 
Since it is related to QChem jobs, could you please review it when you are available? 
Please free feel to ping me if you need anything from my side. Thanks.

**Review by samblau — CHANGES_REQUESTED** [review]
Very minor changes suggested. Once they are made, this should be ready to merge!

File: atomate/qchem/fireworks/tests/test_core.py [hunk 1]
```diff
@@ -132,6 +136,72 @@ def test_SinglePointFW_not_defaults(self):
         self.assertEqual(firework.parents, [])
         self.assertEqual(firework.name, "special single point")
 
+    def test_ProtonEnergyFW_not_defaults(self):
```
[the remark this answers is not in the document]
Comment (rdguha1995) [reply atomate/qchem/fireworks/tests/test_core.py]: Thanks @samblau for the comments! I have implemented them and I think this should be ready to merge. Thanks @Zhuoying for all the help.

**Comment by rdguha1995** [comment]
Hey @Zhuoying! It would be great if you can take some time to merge this. I have implemented all the changes suggested by @samblau

**Comment by rdguha1995** [comment]
Hey @Zhuoying . Just circling back on this. Can you take some time out for merging this?

Thanks a lot!

**Comment by rdguha1995** [comment]
Hey @Zhuoying , can you help me with the current errors?  They are not in codes I have played with. What would be the best way to solve this?

**Comment by Zhuoying** [comment]
@rdguha1995 Yes, the error seems to be induced by the recent change in pymatgen.io.feff (https://github.com/materialsproject/pymatgen/blob/c1e47a78bcf5d4f23b11b83570adb861f47cb183/pymatgen/io/feff/sets.py#L357), not your part. 
I will merge your PR for now. And fix the error in another PR. Thanks!

**Comment by janosh** [comment]
CI fix submitted in #760.

**Pull request merged**